### DISCUSSION TITLE
1.2.26 test is always passing

### DISCUSSION THREAD

**Discussion by yoavrotems** [aquasecurity/kube-bench#659] [Development]
https://github.com/aquasecurity/kube-bench/blob/master/cfg/cis-1.5/master.yaml#L745 
This currently check if the value exist or if its not exist, one of them will always be true.

**Comment by yoavrotems** [comment]
![image](https://user-images.githubusercontent.com/12915002/89523347-5c0f1480-d7eb-11ea-84f2-18f9077b4b4c.png)
@lizrice  I think maybe this test should be manual, since is changing per system, and there isn't any recommendation but just an example of what to set (300).
WDYT?

**Reply by lizrice** [reply]
The test we have is definitely wrong! I agree, as worded this can't be automated and I've raised a ticket about it with CIS. For now perhaps the best thing is to mark it as Manual.

**Reply by davidhay1969** [reply]
Hey @lizrice @yoavrotems interestingly, I've just started seeing `1.2.26` being marked as `[WARN]` having rebuilt my `kube-bench` image from the `main` branch of [kube-bench](https://github.com/aquasecurity/kube-bench.git) showing up as: -

```text
I0526 14:40:09.158954   31290 check.go:110] -----   Running check 1.2.26   -----
I0526 14:40:09.158966   31290 check.go:133] Test marked as a manual test
```

```text
[WARN] 1.2.26 Ensure that the --request-timeout argument is set as appropriate (Automated)
```

```text
== Remediations master ==
1.2.26 Edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
and set the below parameter as appropriate and if needed.
For example,
--request-timeout=300s
```

I'm guessing that this is a result of this [PR](https://github.com/aquasecurity/kube-bench/pull/874/files)

Whilst I can easily configure KB to skip the test via `--skip=1.2.26` I'm wondering whether CIS ever responded, given that their guidelines: -

```text
Description:
Set global request timeout for API server requests as appropriate.
Rationale:
Setting global request timeout allows extending the API server request timeout limit to a duration appropriate to the user's connection speed. By default, it is set to 60 seconds which might be problematic on slower connections making cluster resources inaccessible once the data volume for requests exceeds what can be transmitted in 60 seconds. But, setting this timeout limit to be too large can exhaust the API server resources making it prone to Denial-of-Service attack. Hence, it is recommended to set this limit as appropriate and change the default limit of 60 seconds only if needed.
```

aren't terribly clear 😢 

Could we, for example, test for the existence of `--request-timeout` and, if it's found, throw a warning *IF* it's anything other than 60 i.e. "DANGER, WILL ROBINSON 🤖 " ?

Just a thought ....

**Reply by davidhay1969** [reply]
If so, happy to have a go !
